client/my-sites/sharing/controller.js

```diff
@@ -18,17 +18,24 @@ import { sectionify } from 'lib/route';
 import Sharing from './main';
 import SharingButtons from './buttons/buttons';
 import SharingConnections from './connections/connections';
-import sites from 'lib/sites-list';
-import utils from 'lib/site/utils';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { canCurrentUser, isJetpackModuleActive } from 'state/selectors';
+import { getSiteSettings } from 'state/site-settings/selectors';
+import { requestSiteSettings } from 'state/site-settings/actions';
+import { isJetpackSite, getSiteSlug, getSiteOption } from 'state/sites/selectors';
+import versionCompare from 'lib/version-compare';
 
 const analyticsPageTitle = 'Sharing';
 
 export const layout = context => {
-	const site = sites().getSelectedSite();
 	const { contentComponent, path, store } = context;
+	const state = store.getState();
 
-	if ( site && ! site.settings && utils.userCan( 'manage_options', site ) ) {
-		site.fetchSettings();
+	const siteId = getSelectedSiteId( state );
+	const siteSettings = getSiteSettings( state, siteId );
+
+	if ( siteId && ! siteSettings && canCurrentUser( state, siteId, 'manage_options' ) ) {
+		store.dispatch( requestSiteSettings( siteId ) );
```

Comment:
Issuing requests for data that are needed to render the UI is usually done inside the React components that need them, by rendering `<QuerySiteSettings />`. Is the `requestSiteSettings` really still needed here? The `sharing/buttons/buttons` component uses `QuerySiteSettings` and other component don't seem to use the site settings data.

Reply:
That's a good point. I'll remove this code.